[12.0][MIG] Migration of account_financial_report

wpichler opened the pull request:
I've started with the migration of the account_financial_reporting modul. I was able to get it running on v12 with nearly no changes needed. Just some datetime related things, and one query using parent_left as an id value.
I've removed the menu entries which pointed to the original odoo reports. As far as i can see they are not needed any more - or i am wrong with this ?

Commit:
[FIX] account_financial_report_qweb: date condition in open items report.
This commit includes move lines on the report date by default. These lines shouldn't be considered in the futur.

JordiBForgeFlow committed:
migrate account_financial_report_qweb to 11.0

JordiBForgeFlow committed:
fix issue with currency in general ledger

JordiBForgeFlow committed:
dynamize general ledger

JordiBForgeFlow committed:
remove dependency with account_fiscal_year

oca-transbot committed:
OCA Transbot updated translations from Transifex

rgarnau committed:
[ADD] Responsive reports

Commit:
[10.0] AFR Aged partner balance : columns and datas consistency. (#383)

oca-transbot committed:
OCA Transbot updated translations from Transifex

oca-transbot committed:
OCA Transbot updated translations from Transifex

Commit:
Update links in report, add account group file, update trial balance with hierarchy.

Update indentation, remove empty lines from header.

Update test.

Update pylint.

Remove company_id on computing accounts, since account.group is not a company based model, filtering accounts is done on trial balance report.

Update account variables.

Improve condition in padding on accounts.

Add option to print hierarchy based on defined accounts/computed accounts.

Add VAT report, hierarchy from tax tags ans taxes.

Fix pylint, xlsx report generation header.

Update code to select code_prefix or name.

Update code to select code_prefix or name.

Update code to select code_prefix or name.

Fix domain in base amounts in vat report.

Change trial balance code_prefix or name.

Update trail balance, add tests for vat report.

Update pylint, amounts as monetary, many2one option on generation excels.

Update pulint.

Add VAT Report in readme.

Add VAT Report in readme.

Update array_agg.

Update array_agg.

Update array_agg.

Add option in VAT Report to be printed on Tax Tags - Tax Groups.

Add widget to hierarchy_on on trial balance.

oca-transbot committed:
OCA Transbot updated translations from Transifex

MiquelRForgeFlow committed:
[IMP] account_financial_report: adapt the wizards to multicompany

JordiBForgeFlow committed:
fixes the reporting of unaffected earnings account

oca-transbot committed:
OCA Transbot updated translations from Transifex

HviorForgeFlow committed:
[11.0][IMP] account_financial_report - foreign_currency

oca-transbot committed:
OCA Transbot updated translations from Transifex

sbidoul committed:
remove obsolete .pot files [ci skip]

oca-travis committed:
[UPD] Update account_financial_report.pot

osoul-dev committed:
Added translation using Weblate (Arabic)

osoul-dev committed:
Translated using Weblate (Arabic)

Currently translated at 95.7% (177 of 185 strings)

Translation: account-financial-reporting-11.0/account-financial-reporting-11.0-account_financial_report
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-11-0/account-financial-reporting-11-0-account_financial_report/ar/

osoul-dev committed:
Translated using Weblate (Arabic)

Currently translated at 96.2% (178 of 185 strings)

Translation: account-financial-reporting-11.0/account-financial-reporting-11.0-account_financial_report
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-11-0/account-financial-reporting-11-0-account_financial_report/ar/

BT-astauder committed:
Increasing version because of #PR403

Some changes in the databases been made in #403, but no migration is needed.

pedrobaeza committed:
[FIX] account_financial_report: Fix tests + sql for integration tests

When this module is installed along with other chart account different from generic one,
the number of expected accounts and the computation change (for example, in Spain,
the unaffected earnings account is 129000, choking with group with code prefix 1).

This commit makes the tests resistent to these changes.

Commit:
[ADD] 11.0 account_financial_report: add filter by journals

To general ledger

Commit:
[ADD] 11.0 account_financial_report: journals filter for trial balance

HviorForgeFlow committed:
move font-family from generic tags

oca-travis committed:
[UPD] Update account_financial_report.pot

ons-eco committed:
[11][FIX] Replace partner_ids by journal_ids

p-tombez committed:
[11.0][IMP] Improve transient models cleaning

When there are a lot of account.move.line (several millions) and print any of
the Qweb reports, that will generate also a lot of transient objects.
As these objects are created with an "insert" query, the cleaning normally
triggered by the count of the records in transient tables is not done, so only
the cleaning based on the age of the records is processed (by default, records
older than 1 hours are deleted), but the cron task is only ran one time per
day. For large setups this can lead to memory errors at that point. This change
prevents the memory error by executing the transient record cleanup for the
report models in this module in SQL.

oca-travis committed:
[UPD] Update account_financial_report.pot

rschnapka committed:
Translated using Weblate (German)

Currently translated at 73.9% (215 of 291 strings)

Translation: account-financial-reporting-11.0/account-financial-reporting-11.0-account_financial_report
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-11-0/account-financial-reporting-11-0-account_financial_report/de/

Commit:
Translated using Weblate (German)

Currently translated at 74.2% (216 of 291 strings)

Translation: account-financial-reporting-11.0/account-financial-reporting-11.0-account_financial_report
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-11-0/account-financial-reporting-11-0-account_financial_report/de/

pedrobaeza committed:
Translated using Weblate (Spanish)

Currently translated at 50.2% (146 of 291 strings)

Translation: account-financial-reporting-11.0/account-financial-reporting-11.0-account_financial_report
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-11-0/account-financial-reporting-11-0-account_financial_report/es/

astirpe committed:
[FIX] _transient_clean_rows_older_than() WHERE clause

yaseentai committed:
Translated using Weblate (Arabic)

Currently translated at 96.2% (178 of 185 strings)

Translation: account-financial-reporting-11.0/account-financial-reporting-11.0-account_financial_report
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-11-0/account-financial-reporting-11-0-account_financial_report/ar/

jimmybanegas committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (291 of 291 strings)

Translation: account-financial-reporting-11.0/account-financial-reporting-11.0-account_financial_report
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-11-0/account-financial-reporting-11-0-account_financial_report/es/

jimmybanegas committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (291 of 291 strings)

Translation: account-financial-reporting-11.0/account-financial-reporting-11.0-account_financial_report
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-11-0/account-financial-reporting-11-0-account_financial_report/es/

Commit:
Translated using Weblate (Dutch)

Currently translated at 83.5% (243 of 291 strings)

Translation: account-financial-reporting-11.0/account-financial-reporting-11.0-account_financial_report
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-11-0/account-financial-reporting-11-0-account_financial_report/nl/

Commit:
Translated using Weblate (Spanish)

Currently translated at 100.0% (291 of 291 strings)

Translation: account-financial-reporting-11.0/account-financial-reporting-11.0-account_financial_report
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-11-0/account-financial-reporting-11-0-account_financial_report/es/

alessandrocamilli committed:
Translated using Weblate (Italian)

Currently translated at 23.0% (67 of 291 strings)

Translation: account-financial-reporting-11.0/account-financial-reporting-11.0-account_financial_report
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-11-0/account-financial-reporting-11-0-account_financial_report/it/

JordiBForgeFlow committed:
[account_financial_report] adds the following features to Trial Balance:
- Adds 'Period balance' column
- Renames the option 'Hide accounts at 0'. Means no initial, no debit, credit
  or ending balance for the period.
- Fixes logic to remove lines with 0 activity for the period.

JordiBForgeFlow committed:
[account_financial_report]  improve the calculation in the Trial Balance
of undistributed profits/losses account, and provide a footer note
to the user explaining why will the ending balances will not be zero, but the
period's total profit and loss.

JordiBForgeFlow committed:
add an extra line for P&L application so that the trial balance zeroes in the
balance, and total debits = total credits

JordiBForgeFlow committed:
refactoring of the unaffected earnings reporting in the general ledger
and trial balance.

oca-travis committed:
[UPD] Update account_financial_report.pot

JordiBForgeFlow committed:
remove account groups if hierarchy = 'none'

leemannd committed:
account_financial_report: set backgroud-color to table

psc committed:
Translated using Weblate (German)

Currently translated at 74.6% (217 of 291 strings)

Translation: account-financial-reporting-11.0/account-financial-reporting-11.0-account_financial_report
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-11-0/account-financial-reporting-11-0-account_financial_report/de/

oca-transbot committed:
Update translation files

Updated by Update PO files to match POT (msgmerge) hook in Weblate.

JordiBForgeFlow committed:
[account_financial_report][IMP] Adds the following:
All reports:
- Rename field to hide accounts at 0 to 'hide_account_at_0'
Trial Balance:
- Add possibility to filter by hierarchy levels
- XLSX format will show the hierarchy levels in bold

General Ledger:
- Add the possibility to filter by analytic tags
- Fixes an error on the default date
Journal Ledger:
- The filter on Journals is now optional. If the user does not choose
a journal, by default it will display all journals.

Aged Partner Balance:
- Fixes an error on the default date

Commit:
[FIX] account_financial_report_qweb: Widget date for t-raw date (#483)

pedrobaeza committed:
Translated using Weblate (Spanish)

Currently translated at 96.4% (291 of 302 strings)

Translation: account-financial-reporting-11.0/account-financial-reporting-11.0-account_financial_report
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-11-0/account-financial-reporting-11-0-account_financial_report/es/

JordiBForgeFlow committed:
[account_financial_report] new README structure

dhongu committed:
Translated using Weblate (Romanian)

Currently translated at 39.7% (120 of 302 strings)

Translation: account-financial-reporting-11.0/account-financial-reporting-11.0-account_financial_report
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-11-0/account-financial-reporting-11-0-account_financial_report/ro/

JordiBForgeFlow committed:
[account_financial_report][fix] fixes in usability

OCA-git-bot committed:
[UPD] README.rst

oca-travis committed:
[UPD] Update account_financial_report.pot

oca-transbot committed:
Update translation files

Updated by Update PO files to match POT (msgmerge) hook in Weblate.

primes2h committed:
Translated using Weblate (Italian)

Currently translated at 21.5% (67 of 312 strings)

Translation: account-financial-reporting-11.0/account-financial-reporting-11.0-account_financial_report
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-11-0/account-financial-reporting-11-0-account_financial_report/it/

JordiBForgeFlow committed:
Translated using Weblate (Spanish)

Currently translated at 99.7% (311 of 312 strings)

Translation: account-financial-reporting-11.0/account-financial-reporting-11.0-account_financial_report
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-11-0/account-financial-reporting-11-0-account_financial_report/es/

JordiBForgeFlow committed:
add proper titles

pedrobaeza committed:
[FIX] account_financial_report: Fix account group level computation

Depends was not correct for recomputing when needed + better algorithm

pedrobaeza committed:
[FIX] account_financial_report: Make hide details on 0 work properly

* Passing values to general ledger was stripping some correct records
* Computed field for hiding lines doesn't have proper dependencies nor is not
  taking into account float currency accuracy

OCA-git-bot committed:
[UPD] README.rst

oca-travis committed:
[UPD] Update account_financial_report.pot

oca-transbot committed:
Update translation files

Updated by Update PO files to match POT (msgmerge) hook in Weblate.

JordiBForgeFlow committed:
[FIX] account_financial_report: multicompany+layout behaviour (#498)

* Fix wizards for proper multicompany behaviour.
* Fix layout issue

OCA-git-bot committed:
[UPD] README.rst

oca-travis committed:
[UPD] Update account_financial_report.pot

oca-transbot committed:
Update translation files

Updated by Update PO files to match POT (msgmerge) hook in Weblate.

JordiBForgeFlow committed:
[account_financial_report]
- minor usability fixes for multicompany
- improve performance of general ledger using a new index

OCA-git-bot committed:
[UPD] README.rst

oca-travis committed:
[UPD] Update account_financial_report.pot

oca-transbot committed:
Update translation files

Updated by Update PO files to match POT (msgmerge) hook in Weblate.

JordiBForgeFlow committed:
[account_financial_report] fix title formatting for all reports

wpichler commented:
They tests will fail - because of datetime issues - i will correct them soon

pedrobaeza commented:
Please preserve commit history following technical method explained in https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-12.0

wpichler committed:
[MIG] account_financial_report: Migration to 12.0

pedrobaeza commented:
Please check Travis and runbot status

wpichler commented:
I've fixed most of the problems - most of them were related to the datetime change in odoo 12. Also one problem with the reports (when there is no layout selected - then it will return the wizard to select a layout instead of the report action).
Also installation of the template got changed - hope its correct.
There is still one error in the test_03_partner_balance - here i do not have an idea currently what is failing - maybe someone here has an idea ?

fclementic2c commented:
@wpichler Please ping me as soon as I can do a functional test. 
Thanks for this work.

wpichler commented:
What is about the Profit and Loss and Balance Sheet Reports which are missing in Odoo 12. In Odoo 11 they are still available - are we allowed to take the reports from odoo 11 source and port it to odoo 12 ?

wpichler commented:
@fclementic2c You can already start functional testing - the reports are already ready for testing. Missing are the reports from odoo 11.

pedrobaeza commented:
@wpichler we do it through mis_builder + localization module with the templates.

fclementic2c commented:
It works pretty well ! Good job.
:+1: 

I only have minor comments:

General Ledger: 
tab "Partner Filter" : must not display the partners having a parent_id 
(Odoo 10 fix: https://github.com/OCA/account-financial-reporting/pull/502)
tab "Filter analytic tags" : Error message (if missing group) when I try to select a tag
"
Sorry, you are not allowed to access this document. Only users with the following access level are currently allowed to do that:
- Technical Settings/Analytic Accounting

(Document model: account.analytic.tag)
"
PDF report : problem with header
![image](https://user-images.githubusercontent.com/5735158/51326430-659d2400-1a6f-11e9-831e-dd5ff38aff63.png)

Other reports:

Few M2M fields on wizards have no title which is a bit weird.

Plus, FYI I have recently identified few problems on version 10:
https://github.com/OCA/account-financial-reporting/pull/508
https://github.com/OCA/account-financial-reporting/pull/503
https://github.com/OCA/account-financial-reporting/pull/502
https://github.com/OCA/account-financial-reporting/pull/501
...but it is not merged yet.

wpichler commented:
@pedrobaeza What is the recommended way to integrate the changes in the pull requests by @fclementic2c ? Should i wait until the pull requests are approved - and then cherry pick them ?

pedrobaeza commented:
Yes, because that ones are susceptible to be changed (as there's no agreement on them).

Meanwhile, check Travis and runbot status in this PR.

fclementic2c commented:
@wpichler 
Can you cherry pick PRs above as they are merged now. 
You can ignore #501.

lukebranch commented:
> @wpichler we do it through mis_builder + localization module with the templates.

@pedrobaeza ,

Would you mind pointing me to any localisations in the OCA with P&L and BS built on MIS for reference?

I will need both of these reports for a HK localisation, and can contribute what our team puts together for OCA/l10n-china. Is (@)elicoidal in charge of that repo by the way? I will be developing these for 12.0 for a customer's project.

pedrobaeza commented:
https://github.com/OCA/l10n-spain/tree/11.0/l10n_es_mis_report
https://github.com/OCA/l10n-france/tree/10.0/l10n_fr_mis_reports
https://github.com/OCA/l10n-belgium/tree/11.0/l10n_be_mis_reports

elicoidal commented:
@lukebranch I can help review if necessary.

lukebranch commented:
> https://github.com/OCA/l10n-spain/tree/11.0/l10n_es_mis_report
> https://github.com/OCA/l10n-france/tree/10.0/l10n_fr_mis_reports
> https://github.com/OCA/l10n-belgium/tree/11.0/l10n_be_mis_reports

thanks Pedro.

> @lukebranch I can help review if necessary.

thanks Eric. By the way, should I make a PR to 9.0 in that repo, or would it be acceptable to make a PR to a 12.0 branch, and then backport from there?

> also Switzerland :)
> https://github.com/OCA/l10n-switzerland/tree/11.0/l10n_ch_mis_reports

thanks Frédéric.

fclementic2c commented:
also Switzerland :) 
https://github.com/OCA/l10n-switzerland/tree/11.0/l10n_ch_mis_reports

fclementic2c commented:
I just saw for Odoo12 :  [this](https://github.com/odoo/odoo/commit/b16a40721df1b8812a152de0d155a62bd5d80191)
Now you can have more that 1 account with the type current year earning...which is normal since most charts of accounts have a profit and a loss account.

However, I remember that we added a constraint on these OCA financial reports to block the possibility to have 2 current year earning account.

I am wondering whether report can run correctly if we remove this contraint.

any ideas guys ?

i-vyshnevska commented:
mentioned PR's ported in [512](https://github.com/OCA/account-financial-reporting/pull/512)

osoul-dev commented on account_financial_report/report/trial_balance.py:
```diff
@@ -0,0 +1,577 @@
+    accgroup.parent_id,
+    coalesce(accgroup.code_prefix, accgroup.name),
+    accgroup.name,
+    accgroup.id * 100000,
```
I think there is an issue here. Account group ordering is based on database id, and not on the hierarchy of account groups

pedrobaeza replied:
This is for having unique IDs, so it's correct, unless you have more than 100k accounts.

osoul-dev replied:
No, this is for the sequence field of report_trial_balance_account. Check line 393

pedrobaeza replied:
OK, I see, but this is working in previous versions, so if this hasn't been changed, then it should work through other logic. I have to say that I hate all this SQL stuff, but it's what we have for now, and If you are only checking code and don't have specific issue trying the module, then it's not a bug.

osoul-dev replied:
previous version used parent_left which was okay because the field is sorted based on the hierarchy. Version 12 doesn't have the concept of parent_left/parent_right, and using parent_id won't help either.

The only mitigation I can think of is to depend on the code_prefix for ordering. Not the best option, but for sure much better than the account database id.

osoul-dev replied:
https://github.com/OCA/account-financial-reporting/blob/11.0/account_financial_report/report/trial_balance.py#L404

pedrobaeza replied:
OK, then this has been changed here... Maybe the red CIs is due to these changes. Any way, you still have `parent_path` field for getting ordered records. Why not use it?

I plan to rewrite this for avoiding the SQLs in the future for finally remove this SQL hell, but not sure when.

eLBati replied:
@osoul @pedrobaeza should this be `accgroup.parent_id * 100000`?

eLBati replied:
Considering that in v11 is `accgroup.parent_left * 100000`

pedrobaeza replied:
Now, that doesn't serve, as the parent_id database IDs are also not ordered. What we can try is to use `parent_path` + account name and ordering that way.

eLBati replied:
@pedrobaeza thanks, see 218a0a0149e4c671a65b5763fa3374a4ecb0e74e

pedrobaeza commented:
@wpichler Please check Travis and runbot status.

fclementic2c commented:
@wpichler Do you plan to finish this PR ? thxs

wpichler commented:
I will try to fix the runbot problems today

wpichler commented:
I've tried to fix the problems on the failing tests - but i don't get the point here...

The first failure is here
`        
# Generate the general ledger line
        lines = self._get_report_lines(with_partners=True)
        self.assertEqual(len(lines['partner_receivable']), 0)

        # Add a move at the previous day of the first day of fiscal year
        # to check the initial balance
        self._add_move(
            date=self.previous_fy_date_end,
            receivable_debit=1000,
            receivable_credit=0,
            income_debit=0,
            income_credit=1000
        )

        # Re Generate the general ledger line
        lines = self._get_report_lines(with_partners=True)
        self.assertEqual(len(lines['partner_receivable']), 1)
`

it does generate the general_ledger_report - on first test there should not be a partner_receivable line - then it does add a move at the end of last year - does generate the report - now there should be a partner_receivable line - but this is missing here.

I've tried to find the problem - but i have no real idea where to start. 

Maybe they changes in abstract_test.py are not correct - where it does load the chart template - this would effect all next operations....
I've started a review for this

wpichler commented:
Is it possible to not clean the test data after running the tests ?

i-vyshnevska commented:
data created in https://github.com/OCA/OCB/blob/12.0/addons/account/tests/account_test_classes.py so when you test account related staff db not empty

qgroulard committed:
Add account_financial setup

qgroulard committed:
[REF] update translation

qgroulard committed:
[FIX] Few fixes to solve warnings:

- _description in account_financial_report_abstract
- _description in vat.report.wizard
- Two fields using same label 'Accounts' in account.group
- Two fields using same label 'Account' in report_journal_ledger_move_line
- Two fields using same label 'Partner' in report_journal_ledger_move_line
- Two fields using same label 'Accounts' in report_trial_balance_account
- Two fields using same label 'Tax' in report_vat_report_taxtag
- Two fields using same label 'Tax' in report_vat_report_tax
- Two fields using same label 'Filter accounts' in general.ledger.report.wizard

wpichler committed:
Merge pull request #1 from acsone/12.0-account_financial_setup_add_setup-qgr

[12.0] [ADD] Add account_financial setup

wpichler committed:
Merge pull request #2 from acsone/12.0-mig-account_financial_report_fix_warning-qgr

[12.0] [FIX] Few fixes to solve warnings:

wpichler committed:
Merge pull request #3 from acsone/12.0-account_financial_report_update_translation-qgr

[12.0] [REF] update translation

MiquelRForgeFlow commented:
that's a wrong rebase

wpichler commented:
@mreficent What do you mean with wrong rebase ?

MiquelRForgeFlow commented:
I mean that it seems to me that this PR has not been rebased correctly or it's pending to do a rebase. To do a proper rebase, please follow this:

```
git checkout your-pr-branch
git fetch oca-remote 12.0
git rebase oca-remote/12.0 
```

:)

stevech091 reviewed (commented):
funcional ok

stevech091 reviewed (approved):
ok

wpichler commented:
@mreficent ah ok - i will try to fix some travis errors - and then do a new rebase

AngelCamacho commented:
anyone knows what we need to merge and close the issue?

pedrobaeza commented:
@AngelCamacho it needs reviews + put travis green

wpichler committed:
[IMP] Reformated one query to avoid one sql injection warning message - but no sql injection was possible here

wpichler commented:
I've now done a new rebase + fixed one travis sql injection warning (which was only a warning - no sql injection was possible here).
The other travis error's about sql injection are to ignore - no sql injection is possible here. There are statements in the form:

```
basic_where_clause = 'blah'
sql_query = 'SELECT 1 FROM table ' + basic_where_clause

```

So we can ignore them

eLBati commented:
@wpichler flake8 is failing with
```
account_financial_report/wizard/trial_balance_wizard.py:13:1: E302 expected 2 blank lines, found 1
account_financial_report/wizard/trial_balance_wizard.py:99:80: E501 line too long (95 > 79 characters)
account_financial_report/wizard/general_ledger_wizard.py:16:1: E302 expected 2 blank lines, found 1
account_financial_report/wizard/general_ledger_wizard.py:91:80: E501 line too long (95 > 79 characters)
account_financial_report/report/trial_balance.py:203:80: E501 line too long (91 > 79 characters)
account_financial_report/report/general_ledger.py:227:80: E501 line too long (91 > 79 characters)
account_financial_report/report/open_items.py:163:80: E501 line too long (91 > 79 characters)
account_financial_report/report/abstract_report_xlsx.py:194:80: E501 line too long (85 > 79 characters)
account_financial_report/report/journal_ledger.py:597:80: E501 line too long (91 > 79 characters)
account_financial_report/tests/test_general_ledger.py:5:1: F401 'time' imported but unused
account_financial_report/tests/test_general_ledger.py:91:80: E501 line too long (82 > 79 characters)
account_financial_report/tests/test_general_ledger.py:92:80: E501 line too long (81 > 79 characters)
account_financial_report/tests/test_journal_ledger.py:4:1: F401 'time' imported but unused
account_financial_report/tests/test_vat_report.py:160:80: E501 line too long (87 > 79 characters)
account_financial_report/tests/test_vat_report.py:278:80: E501 line too long (89 > 79 characters)
account_financial_report/tests/test_vat_report.py:291:80: E501 line too long (89 > 79 characters)
account_financial_report/tests/test_trial_balance.py:5:1: F401 'time' imported but unused
```

About lint, if there is not SQL injection risk, you can add `# noqa` to the failing lines.

Also, tests are failing with
```
2019-05-10 16:13:21,360 5482 ERROR openerp_test odoo.addons.account_financial_report.tests.test_general_ledger: `   File "/home/travis/build/OCA/account-financial-reporting/account_financial_report/tests/test_general_ledger.py", line 291, in test_02_partner_balance
2019-05-10 16:13:21,360 5482 ERROR openerp_test odoo.addons.account_financial_report.tests.test_general_ledger: `     self.assertEqual(len(lines['partner_receivable']), 1)
2019-05-10 16:13:21,360 5482 ERROR openerp_test odoo.addons.account_financial_report.tests.test_general_ledger: ` AssertionError: 0 != 1
```

eLBati reviewed (changes requested):
Also, it seems this issue https://github.com/OCA/account-financial-reporting/pull/509#pullrequestreview-206896535 is still open

And changes from https://github.com/OCA/account-financial-reporting/pull/541 are not present

eLBati commented:
An issue related to new dates format (tested in trial balance):

![trial-bal](https://user-images.githubusercontent.com/1033131/58337566-5d525c80-7e46-11e9-913a-3c3e0115fb3e.gif)

In particular, the `domain` here
```
            return this.do_action({
                type: 'ir.actions.act_window',
                res_model: res_model,
                domain: domain,
                views: [[false, "list"], [false, "form"]],
                target: 'current'
            });
```
is like `[('account_id', 'in', [27, 186]), ('date', '<', datetime.date(2019, 5, 1))]`
while I think it should be `[('account_id', 'in', [27, 186]), ('date', '<', '2019-05-01')]`

@wpichler let me know whether you plan to work on this PR or you need help.
Thanks!

wpichler commented:
@eLBati Help would be fine

pedrobaeza commented:
Please include #512 and #544

MiquelRForgeFlow commented:
Also https://github.com/OCA/account-financial-reporting/pull/514.

MiquelRForgeFlow commented:
Also https://github.com/OCA/account-financial-reporting/pull/554 and https://github.com/OCA/account-financial-reporting/pull/555.

eLBati commented:
Moving to https://github.com/OCA/account-financial-reporting/pull/559

pedrobaeza commented:
@wpichler I hope you are not upset for the superseeding, but we want to finish this and as you were not answering lately for the cherry-pickings, it's better to move ahead. Your attribution have been respected any way.

Thanks for the comprehension.
